[docs] Document the supported formats (#8746)
mui/mui-x: docs/data/date-pickers/adapters-locale/adapters-locale.md

You can control the field format spacing using the [formatDensity](/x/react-date-pickers/custom-field/#change-the-format-density) prop.
:::

> #### Supported formats

Reviewer: Have you considered h3 instead? Then this section would be included in the side nav 🤔 
```suggestion
### Supported formats
```

Author: I moved to h3 and renamed the title since it's no longer a child of it's previous parent

Some format might not yet be supported by the fields.
For example, they don't support day of the year or quarter.

> Here is the list of the formats currently supported:

Reviewer: ```suggestion
Here is the list of the currently supported formats:
```

- The year
  - ✅ 2-digits values (e.g: `23`)
  - ✅ 4-digits values (e.g: `2023`)

>   - ❌ Values with ordinal (e.g: `2023th`)

Reviewer: For some reason, at least on MacOS the x is not presented in red color when rendered 🙈 🤷 
Do you have any idea what could be wrong? 🤔 
<img width="463" alt="Screenshot 2023-04-26 at 10 09 50" src="https://user-images.githubusercontent.com/4941090/234497555-33d0b455-8429-4e33-a409-4aa5f9a541a4.png">

Author: Same on Linux
If you have a better icon I'm Interested, I took the one of [this page](https://mui.com/x/react-data-grid/getting-started/#feature-comparison) but it's a little sad in grey

Reviewer: IMHO, in that particular page, I don't mind the grey color as it reduces the noise, and when on a table, the idea is easier to grasp without a clear color difference. 🤔

Reviewer: I've narrowed the reason behind this styling: The `fontFamily` is to blame. Removing the first `"IBM Plex Sans"` entry results in a red cross. 🙈

- The month

  - ✅ 1-based digit (e.g: `08`)
  - ✅ Multi-letter values (e.g `Aug`, `August`)
  - ❌ 1-letter values (e.g: `A`) because several months are represented with the same letter

- The day of the month

  - ✅ 1-based digit values (e.g: `24`)
  - ✅ 1-based digit values with ordinal (e.g: `24th`)

- The day of the week

  - ✅ 0-based digit values (e.g: `03`)
  - ✅ 1-based digit values (e.g: `04`)
  - ✅ Multi-letter values (e.g: `Tue`, `Tuesday`)
  - ❌ 1-letter values (e.g: `T`) because several days of the week are represented with the same letter

- The hours

  - ✅ 0-based 12-hours values (e.g: `03`)
  - ✅ 0-based 24-hours values (e.g: `15`)
  - ❌ 1-based values (e.g: `24` instead of `00`)

- The minutes

- The seconds

- The meridiem

> If you need to use some formats that is not yet compatible, please [open an issue](https://github.com/mui/mui-x/issues/new/choose) describing what is your exact use case.

Reviewer: ```suggestion
If you need to use some format that is not yet supported, please [open an issue](https://github.com/mui/mui-x/issues/new/choose) describing what is your exact use case.
```
